File: 0053. Maximum Subarray/Solution.c
#include <limits.h>
#include <math.h>

int maxSubArray(int* nums, int numsSize)
{
    int res = INT_MIN, cur = 0;
    for (int i = 0; i < numsSize; i++) {
        if (cur < 0)
            cur = nums[i];
        else
            cur += nums[i];
        res = fmax(res, cur);
    }
    return res;
}

File: 0053. Maximum Subarray/Solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int res = Integer.MIN_VALUE, cur = 0;
        for (int e : nums) {
            if (cur < 0) {
                cur = e;
            } else {
                cur += e;
            }
            res = Math.max(res, cur);
        }
        return res;
    }
}
